systems-manager.tf

```diff
@@ -0,0 +1,98 @@
+# AWS KMS alias used for encryption/decryption of SSM secure strings
+variable "kms_alias_name_ssm" {
+  type        = string
+  default     = "alias/aws/ssm"
+  description = "KMS alias name for SSM"
+}
+
+variable "ssm_enabled" {
+  type        = bool
+  default     = false
+  description = "If `true` create SSM keys for the database user and password."
+}
+
+variable "ssm_key_format" {
+  type        = string
+  default     = "/%v/%v/%v"
+  description = "SSM path format. The values will will be used in the following order: `var.ssm_key_prefix`, `var.database_name`, `var.ssm_key_*`"
+}
+
+variable "ssm_key_prefix" {
+  type        = string
+  default     = "rds"
+  description = "SSM path prefix. Omit the leading forward slash `/`."
+}
+
+variable "ssm_key_user" {
+  type        = string
+  default     = "admin/db_user"
+  description = "The SSM key to save the user. See `var.ssm_path_format`."
+}
+
+variable "ssm_key_password" {
+  type        = string
+  default     = "admin/db_password"
+  description = "The SSM key to save the password. See `var.ssm_path_format`."
+}
+
+variable "ssm_key_hostname" {
+  type        = string
+  default     = "admin/db_hostname"
+  description = "The SSM key to save the hostname. See `var.ssm_path_format`."
+}
+
+variable "ssm_key_port" {
+  type        = string
+  default     = "admin/db_port"
+  description = "The SSM key to save the port. See `var.ssm_path_format`."
+}
+
+locals {
+  ssm_enabled = local.enabled && var.ssm_enabled
+}
+
+resource "aws_ssm_parameter" "rds_database_user" {
+  count = local.ssm_enabled ? 1 : 0
+
+  name        = format(var.ssm_key_format, var.ssm_key_prefix, var.database_name, var.ssm_key_user)
```

Comment:
There are cases where `database_name` needs to be `null` for the SQL server. 

Would it be acceptable to replace `var.database_name` here with `module.this.id` ALWAYS ? or would it be better to only replace it with `module.this.id` when `var.database_name == null` or would that be confusing ?

Open to thoughts here.

Reply:
We're going to use `module.this.name` in place of `var.database_name`